src/System.Management.Automation/engine/CommandProcessorBase.cs

```diff
@@ -169,24 +169,38 @@ internal bool UseLocalScope
             InvocationInfo invocationInfo)
         {
             // If we are in a constrained language mode (Core or Restricted), block it.
-            // This goes both ways:
+            // We are currently restricting in one direction:
             //    - Can't dot something from a more permissive mode, since that would probably expose
             //      functions that were never designed to handle untrusted data.
-            //    - Can't dot something from a less permissive mode, since that might introduce tainted
-            //      data into the current scope.
```

Comment:
>This code won't run for NoLanguage mode so the only direction checked is trusted (full language mode) script running in a constrained/restricted session.

I see. Can you please add this in the comment? I think that would be easier for people to understand the comment and the code.